pl_bolts/models/gans/dcgan/dcgan_module.py

```diff
@@ -0,0 +1,228 @@
+
+        return disc_loss
+
+    def _get_gen_loss(self, real: torch.Tensor) -> torch.Tensor:
+        # Train with fake
+        fake_pred = self._get_fake_pred(real)
+        fake_gt = torch.ones_like(fake_pred)
+        gen_loss = self.criterion(fake_pred, fake_gt)
+
+        return gen_loss
+
+    def _get_fake_pred(self, real: torch.Tensor) -> torch.Tensor:
+        batch_size = len(real)
+        noise = self._get_noise(batch_size, self.hparams.latent_dim)
+        fake = self(noise)
+        fake_pred = self.discriminator(fake)
+
+        return fake_pred
+
+    def _get_noise(self, n_samples: int, latent_dim: int) -> torch.Tensor:
+        return torch.randn(n_samples, latent_dim, device=self.device)
+
+    @staticmethod
+    def add_model_specific_args(parent_parser: ArgumentParser) -> ArgumentParser:
+        parser = ArgumentParser(parents=[parent_parser], add_help=False)
+        parser.add_argument("--beta1", default=0.5, type=float)
+        parser.add_argument("--feature_maps_gen", default=64, type=int)
+        parser.add_argument("--feature_maps_disc", default=64, type=int)
+        parser.add_argument("--image_channels", default=1, type=int)
+        parser.add_argument("--latent_dim", default=100, type=int)
+        parser.add_argument("--learning_rate", default=0.0002, type=float)
+        return parser
+
+
+def cli_main(args=None):
+    pl.seed_everything(1234)
+
+    parser = ArgumentParser()
+    parser.add_argument("--dataset", default="mnist", type=str, choices=["cifar10", "mnist"])
+    parser.add_argument("--image_size", default=64, type=int)
+    parser.add_argument("--data_dir", default="./", type=str)
+    parser.add_argument("--batch_size", default=64, type=int)
+    parser.add_argument("--num_workers", default=6, type=int)
+
+    script_args, _ = parser.parse_known_args(args)
+
+    if script_args.dataset == "cifar10":
+        transforms = transform_lib.Compose(
+            [
+                transform_lib.Resize(script_args.image_size),
+                transform_lib.ToTensor(),
+                transform_lib.Normalize((0.5, 0.5, 0.5), (0.5, 0.5, 0.5)),
+            ]
+        )
+        dataset = CIFAR10(root=script_args.data_dir, download=True, transform=transforms)
+    elif script_args.dataset == "mnist":
+        transforms = transform_lib.Compose(
+            [
+                transform_lib.Resize(script_args.image_size),
+                transform_lib.ToTensor(),
+                transform_lib.Normalize((0.5,), (0.5,)),
+            ]
+        )
+        dataset = MNIST(root=script_args.data_dir, download=True, transform=transforms)
+
+    dataloader = DataLoader(
+        dataset, batch_size=script_args.batch_size, shuffle=True, num_workers=script_args.num_workers
+    )
+
+    parser = pl.Trainer.add_argparse_args(parser)
+    parser = DCGAN.add_model_specific_args(parser)
+    args = parser.parse_args(args)
+
+    model = DCGAN(**vars(args))
+    callbacks = [
```

Comment:
Not sure if we implement callbacks in all models, we should probably have consistency

Reply:
Do you mean that we should get rid of callbacks from `cli_main()` of all models?

Reply:
Since we don't use TensorBoard Logging in all models it might be odd to support just one model. Hence I thuoght we can probably have consistency.

Reply:
I see. For `cli_main()` of this model, I guess it's useful to have `TensorboardGenerativeModelImageSampler` to see generated images by just running `python dcgan_module.py --gpus 1` in my humble opinion. However, for other general callbacks like `ModelCheckpoint`, I guess we should be consistent as you mentioned...

cc: @Borda